internal/rbd/rbd_attach.go

```diff
@@ -235,13 +235,17 @@ func createPath(ctx context.Context, volOpt *rbdVolume, cr *util.Credentials) (s
 	if volOpt.readOnly {
 		mapOptions = append(mapOptions, "--read-only")
 	}
+
+	if volOpt.MapOptions != "" {
+		mapOptions = append(mapOptions, "--options", volOpt.MapOptions)
```

Comment:
if NBD is used, this will not work. It needs to be checked.

Reply:
Any idea why it won't work?

Reply:
`rbd-nbd map ...` does not seem to have a `--options` parameter. The arguments for kRBD `rbd map` are just different.

Reply:
> if NBD is used, this will not work. It needs to be checked.

I don't think so.  `rbd device map -t nbd` accepts nbd-specific options (and even if it didn't, we are checking for `""`).  `mapOptions` and `unmapOptions` are low-level, if anything is specified we want them passed through unconditionally.

Reply:
> `rbd-nbd map ...` does not seem to have a `--options` parameter.

Correct, but `rbd device map -t nbd --options foo,bar ...` would be translated into `rbd-nbd map --foo --bar ...`.

Reply:
Ah, ok! Thank for clarifying. `--device-type` is used with the `rbd` command, and `rbd-nbd` is executed as a child process.